pyxrf/model/command_tools.py

```diff
@@ -208,40 +223,192 @@ def pyxrf_batch(start_id, end_id=None, wd=None, fit_channel_sum=True, param_file
+    Files must have extension .h5. (Note, that ``.`` separating the extension is a non-number character
+    that may terminate the Scan ID part of the file name
+
+    For example, the following files will be recognized as Scan ID 28355:
+
+        scan2D_28355.h5
+        scanNewName_28355_some_comments.h5
+        scan_2D_28355.h5  (incorrect, recognized as Scan ID 2, not 28355)
+        scan2D28355.h5 (incorrect, no ``_``)
+
+    Examples
+    --------
+
+    In the following examples it is assumed that all .h5 files and the .json parameter file are
+    located in the current directory.
+
+    -- Process data file with Scan ID 455.
+
+        .. code: python
+
+            pyxrf_batch(455, param_file_name="parameters.json")
+
+    -- Process data files with Scan IDs in the range 455 .. 457:
+
+        .. code: python
+
+            pyxrf_batch(455, 457, param_file_name="parameters.json")
+
+    -- Process all data files in the current directory:
+
+        .. code: python
+
+            pyxrf_batch(param_file_name="parameters.json")
+
     """
+
+    print()
+
     if wd is None:
         wd = '.'
-    all_files = glob.glob(os.path.join(wd, '*.h5'))
-
-    if end_id is None:
-        flist = [fname for fname in all_files if str(start_id) in fname]
-        try:
-            fpath = flist[0]
-        except IndexError:
-            print("File with runid {} doesn't exist.".format(start_id))
-        fname = fpath.split(sep_v)[-1]
-        working_directory = fpath[:-len(fname)]
-        fit_pixel_data_and_save(working_directory, fname,
-                                fit_channel_sum=fit_channel_sum, param_file_name=param_file_name,
-                                fit_channel_each=fit_channel_each, param_channel_list=param_channel_list,
-                                incident_energy=incident_energy, spectrum_cut=spectrum_cut,
-                                save_txt=save_txt, save_tiff=save_tiff,
-                                ic_name=ic_name, use_average=use_average)
     else:
+        wd = os.path.expanduser(wd)
+    param_file_name = os.path.expanduser(param_file_name)
+
+    if data_files is not None:
+
+        # Check if ``data_files`` has valid value
+        data_files_valid = False
+        if isinstance(data_files, str):
+            data_files = [data_files]  # Convert to list
+            data_files_valid = True
+        elif isinstance(data_files, Iterable):
+            data_files = list(data_files)  # Convert to list
+            data_files_valid = True
+            for fln in data_files:
+                if not isinstance(fln, str):
+                    data_files_valid = False
+                    break
+
+        if not data_files_valid:
+            print(f"ERROR: the list of data files has invalid format. Check the parameter 'data_files'.")
+            print(f"   data_files = {data_files}")
+            raise ValueError("Function pyxrf_batch. Parameter 'data_files' has invalid format.")
```

Comment:
Thanks. I don't think we need prints anymore as the string can be added to the `ValueError`.

Reply:
I am not sure. Typical way to deal with exceptions in home-made script is to catch them and do nothing. It is easier to for the user to find the cause of error.